config/routes.rb

```diff
@@ -59,6 +59,8 @@
   delete ':collection/:id', to: router
   delete ':collection', to: router
 
+  post ':collection/inbox' => 'resources#index'
```

Comment:
It's not specific to inbox, it will be used every time we want to search now
```suggestion
  post ':collection/search' => 'resources#index'
```